.github/workflows/publish_xgboost.yml

```diff
@@ -0,0 +1,38 @@
+name: xgboost artifact
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [ main ]
+
+permissions:
+  pull-requests: write
+  contents: write
+  repository-projects: write
+
+jobs:
+  publich_xgboost_pkg:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+        fetch-depth: 0
+    - name: Build xgboost
+      run: |
+          set -x
+          export XGBOOST_TAG=$(cat xgboost_tag)
+          sudo apt-get update -y
+          # ubuntu 20.04 ships with cmake 3.16, but xgboost needs 3.18+. Install cmake from pip
+          sudo apt-get install git make g++ python3-pip -y
+          sudo pip install cmake --upgrade
+          git clone --recursive https://github.com/dmlc/xgboost.git
+          cd xgboost && git checkout tags/v${XGBOOST_TAG} && mkdir -p build && cd build && cmake .. && make -j4 && make package && cp xgboost-${XGBOOST_TAG}-Linux.sh /tmp/xgboost.sh
+    - if: startsWith(github.event.ref, 'refs/tags/')
+      name: Release
+      uses: softprops/action-gh-release@master
```

Comment:
can we release file in two gha?

Reply:
Since they are in different GHA, they should be released separately.

Reply:
@rootfs , I suppose as https://github.com/softprops/action-gh-release will only create one release page per once time. Which means, for each time, there will be two release. Once for bcc and one for xgboost?

Reply:
further question as do we just build xgboost for CI usage but not our final build?
if yes, we can just update current GHA with xgboost, as we just run this GHA for one time and for CI or internal usage only.

Reply:
fixed